### DISCUSSION TITLE
Create bottle from local Formula

### DISCUSSION THREAD

**Discussion by a4z** [Homebrew/discussions#4468] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.15
ORIGIN: https://github.com/Homebrew/brew
HEAD: 763c41f006ab9b44261e313b0aaa24a047431b06
Last commit: 7 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: f8376785ae664ea95046c2119f06152bdcc1c80f
Core tap last commit: 9 weeks ago
Core tap branch: master
Core tap JSON: 01 May 13:00 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.40.1 => /opt/homebrew/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 13.3.1-arm64
CLT: 14.3.0.0.1.1679647830
Xcode: 14.3 => /Applications/Xcode-14.3.app/Contents/Developer
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /opt/homebrew/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/opt/homebrew/sbin:$PATH"' >> ~/.zshrc
```

### Description of issue

I try to create a local formula, 
I can build and install it, even run the test, 

but how do I create a bottle that I can later upload together with the formula on some private repo?

I try this

```sh
 brew install --formula --build-bottle ./Formula/my_formular.rb
 brew bottle  Formula/my_formular.rb
```
but that results in

```sh
Error: Formula not installed or up-to-date: my_formular
```

What do I miss? 

Are there any docs for that workflow? The docs I found seem to be about working on homebrew_core, but seem not to cover my local dev attempt

**Comment by carlocab** [comment]
Make sure `Formula/my_formular.rb` lives in a Homebrew tap (e.g. one created with `brew tap-new`), and refer to it by name instead of using a relative path. For example:
```
brew install --formula --build-bottle a4z/your-tap-name/my_formular
brew bottle a4z/your-tap-name/my_formular
```

**Comment by a4z** [comment]
Ok, so I start with `brew tap-new` and place my formula there? 

Do I need to upload it for being able to work with it or can I stay totally local until everything is ready, tested, and can be released?

**Reply by carlocab** [reply, marked as answer]
> Ok, so I start with `brew tap-new` and place my formula there?

Yes, assuming this is a formula you intend to distribute in your own tap.

> Do I need to upload it for being able to work with it or can I stay totally local until everything is ready, tested, and can be released?

Everything can stay local while you're working on it.

**Reply by a4z** [reply]
Ok, I see.
I think I know now what I need to learn upfront.
Thanks a lot for your help!
